lib/rspec/mocks/error_generator.rb

```diff
@@ -291,6 +303,19 @@ def format_received_args(*args)
         args.empty? ? "(no args)" : "(" + received_arg_list(*args) + ")"
       end
 
+      def format_unexpected_with_args(*args)
+        with_args = Hash[Array(args.first).group_by { |x| x }.map { |k, v| [k, v.count] }]
+        received_with_arg_list(*with_args)
+      end
+
+      def received_with_arg_list(*args)
+        list = args.map do |a, i|
+          times_count = args.size > 1 ? '(' + times(i) + ')' : ''
+          ["(#{received_arg_list(*a)})", "#{times_count}"].join(" ")
```

Comment:
I initially had the code setup just like that, but then single expectations would look like this
    
    expected: (:three, :four)
         got: (:one, :four) (1 time)

vs as it is now

    expected: (:three, :four)
         got: (:one, :four)

I'm okay switching it to the one-liner, or maybe adjusting the `times` method to accept an option for ignoring excluding the text if it's 1 time.

As for `format_received_args `,  I'm usually in favor of splitting out methods more then combining, but perhaps this is a situation where we can make it into one more efficient method. I'll see where things lead after I do some of the refactoring on the raise method.